python-sdk/pachyderm_sdk/api/pps/extension.py

```diff
@@ -148,3 +155,58 @@ def create_secret(
         ).encode()
 
         return super().create_secret(file=file)
+
+    def generate_datum(self, input_spec: "Input", batch_size: int) -> Iterator["DatumInfo"]:
+        """Generates datums for a given input spec.
```

Comment:
You could make this a bit more explicit. Something like "Creates a generator that yields batches of datums for a given input spec"